src/utils/trusts.ts

```diff
@@ -318,25 +318,134 @@ const saveIndividualTrusteeInTrust = (trust: Trust, trusteeData: IndividualTrust
 };
 
 /**
- * The Trustee Ids are NOT part of the OE API data model nor part of the SDK mapper object. They need
- * to be generated when the Trust Data is got from the API (e.g. for a "Save and Resume" journey)
- * @param appData - application data
+ * The API returns data in a different type of resource as is sent up to it! This is hard coded in the API
+ * with the Node SDK just doing some date mapping.
+ *
+ * Rather than re-writing the Node SDK and the input part of the API we have this function.
+ *
+ * For primatives no mapping is required, but note:
+ * 1.) The API address structure is different from the Web Address structure and for individual and corporate trustees the
+ * API sends the data back in an API address (even though it received them in flat string fields)
+ * 2.) yesNoResponse is not an issue and while the JSON is true or false from the API (rather 1 or 0) this works fine and is
+ * how things are outside of the trust data
+ * 3.) enums need to be converted expicitly (see example below with the `type` property)
+ * 4.) the API model does not have trustee id's (and these are not added in the Node SDK) so they are re-generated here
+ *
+ * This needs to be called when the Trust Data is got from the API (e.g. for a "Save and Resume" journey)
+ *
+ * @param appData  - application data
  * @returns void
  */
-const generateTrusteeIds = (appData: ApplicationData) => {
+const mapTrustApiReturnModelToWebModel = (appData: ApplicationData) => {
 
   if (containsTrustData(getTrustArray(appData))) {
 
     for (const trust of appData.trusts ?? []) {
 
-      trust.CORPORATES = (trust.CORPORATES as TrustCorporate[]).map( te => {return { ...te, id: uuidv4() }; } );
-      trust.HISTORICAL_BO = (trust.HISTORICAL_BO as TrustHistoricalBeneficialOwner[]).map( te => {return { ...te, id: uuidv4() }; } );
-      trust.INDIVIDUALS = (trust.INDIVIDUALS as TrustIndividual[]).map( te => {return { ...te, id: uuidv4() }; } );
+      trust.CORPORATES = (trust.CORPORATES || []).map(corporateTrustee => {
+
+        const apiData: any = corporateTrustee;
+
+        return {
+          id: uuidv4(),
+          type: getRoleWithinTrustType(apiData.type) as RoleWithinTrustType,
+          name: apiData.name,
+          date_became_interested_person_day: apiData?.date_became_interested_person_day,
+          date_became_interested_person_month: apiData?.date_became_interested_person_month,
+          date_became_interested_person_year: apiData?.date_became_interested_person_year,
+          ro_address_premises: apiData.registered_office_address.property_name_number,
+          ro_address_line_1: apiData.registered_office_address.line_1,
+          ro_address_line_2: apiData?.registered_office_address.line_2,
```

Comment:
noticed that some of the apiData have the null check '?', just wondering if they should all be the same

Reply:
I have copied the null check '?' from the web models in src/model/trust.model.ts (TrustCorporate etc)

Reply:
i think it would be safer to use the ? on all the fields during the mapping, to prevent null pointer errors